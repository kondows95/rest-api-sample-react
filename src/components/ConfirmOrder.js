import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Grid, Box, TextField, Paper, Checkbox, Radio, Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle } from '@material-ui/core';
import CheckoutOrderReviewCart from '../containers/CheckoutOrderReviewCart';
import CheckoutPayment from '../containers/CheckoutPayment';
import { withRouter } from 'react-router-dom'
const useStyles = makeStyles(theme => ({
    form: {
        width: '100%', // Fix IE 11 issue.
        marginTop: theme.spacing(1),
    },
    submit: {
        margin: theme.spacing(3, 0, 2),
    },
    button: {
        margin: theme.spacing(2)
    }
}));

const ConfirmForm = ({ cart, requestParams, postResultObj, setRequestParams, postOrder, history }) => {
    const classes = useStyles()
    console.log(requestParams);
    console.log(postResultObj);
    const [open, setOpen] = React.useState(false);
    const handlePrevious = () => {
        history.push("/checkout");
    };

    const handleSubmit = event => {
        event.preventDefault();
        postOrder();
    }

    const handleHome = event => {
        event.preventDefault();
        history.push("/items");
    }

    const handleClose = () => {
        setOpen(false);
        history.push("/");
    };

    return (
        <Box>

            {postResultObj === null ?

                <form className={classes.form} noValidate onSubmit={handleSubmit}>
                    <Box fontSize="h6.fontSize" pl={3} mt={2}>
                        1.Order contents
                    </Box>
                    <Box flexGrow={1} p={2}>
                        <CheckoutOrderReviewCart />
                    </Box>
                    <Box fontSize="h6.fontSize" pl={3} mt={2}>
                        2.Shipping Address
                    </Box>
                    <Box flexGrow={1} p={2}>
                        <Grid container  >
                            <Grid item xs={12} sm={6}  >
                                <Box p={1} >
                                    <Address label="First Name" value={requestParams !== null ? requestParams.first_name : null} />
                                </Box>
                            </Grid>
                            <Grid item xs={12} sm={6}>
                                <Box p={1} >
                                    <Address label="Last Name" value={requestParams !== null ? requestParams.last_name : null} />
                                </Box>
                            </Grid>
                        </Grid>
                        <Grid container  >
                            <Grid item xs={12} sm={6}  >
                                <Box p={1} >
                                    <Address label="Address1" value={requestParams !== null ? requestParams.address1 : null} />
                                </Box>
                            </Grid>
                            <Grid item xs={12} sm={6}>
                                <Box p={1}>
                                    <Address label="Address2" value={requestParams !== null ? requestParams.address2 : null} />
                                </Box>
                            </Grid>
                        </Grid>
                        <Grid container  >
                            <Grid item xs={12} sm={6}  >
                                <Box p={1} >
                                    <Address label="Country" value={requestParams !== null ? requestParams.country : null} />
                                </Box>
                            </Grid>
                            <Grid item xs={12} sm={6}>
                                <Box p={1}>
                                    <Address label="State" value={requestParams !== null ? requestParams.state : null} />
                                </Box>
                            </Grid>
                        </Grid>
                        <Grid container  >
                            <Grid item xs={12} sm={6}  >
                                <Box p={1} >
                                    <Address label="City" value={requestParams !== null ? requestParams.city : null} />
                                </Box>
                            </Grid>
                        </Grid>
                    </Box>
                    <Box fontSize="h6.fontSize" pl={3} mt={2}>
                        3.Payment Method
                    </Box>
                    <Box flexGrow={1} p={2}>
                        <Paper>
                            <Box p={2} bgcolor="#e9e9ef">Credit Card Pay</Box>
                        </Paper>
                    </Box>
                    <Box mx={3} p={1} mt={1}>
                        <Box textAlign="center"  >
                            <Button onClick={handlePrevious} variant="contained" color="primary" className={classes.button}>
                                Go Previous
                        </Button>
                            <Button type="submit" variant="contained" color="secondary" className={classes.button}>
                                Make an order
                        </Button>
                        </Box>
                    </Box>
                </form>

                :

                <Grid>
                    <Box fontSize="h6.fontSize" pl={3} mt={2}>
                        Thank you for your order.
                    </Box>
                    <Box pl={3} pt={1}>
                        Thank you very much for your order.
                    </Box>

                    <Box pl={3} pt={3}>
                        Order contents have been sent through [order confirmation mail].
                    </Box>
                    <Box pl={3}>
                        Please confirm the order contents, if there are any mismatches or unclear points, please contact us.
                    </Box>
                    <Box flexGrow={1} p={2}>
                        <Paper>
                            <Box p={2} bgcolor="#e9e9ef">Order Number</Box>
                            <Box p={3} mb={3} bgcolor="#fff">{postResultObj.id}</Box>
                        </Paper>
                    </Box>

                    <Box flexGrow={1} p={2}>
                        <Paper>
                            <Box p={2} bgcolor="#e9e9ef">Total Price</Box>
                            <Box p={3} mb={3} bgcolor="#fff">{postResultObj.total_price} MMK</Box>
                        </Paper>
                    </Box>
                    <Box textAlign="center" >
                        <Button onClick={handleHome}  variant="contained" color="secondary" className={classes.button}>
                            Go To HOME
                        </Button>
                    </Box>

                </Grid>
            }
        </Box>
    );
}
export default withRouter(ConfirmForm);

const Address = ({ label, value }) => (
    <Paper>
        <Box p={2} bgcolor="#e9e9ef">{label}</Box>
        <Box p={3} mb={3} bgcolor="#fff">{value}</Box>
    </Paper>
)


