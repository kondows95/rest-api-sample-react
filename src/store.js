import rootReducer from './modules/index';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
export default createStore(rootReducer, applyMiddleware(thunk));