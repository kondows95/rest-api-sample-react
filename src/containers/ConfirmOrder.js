import { connect } from 'react-redux'
import ConfirmOrder from '../components/ConfirmOrder'
import { setRequestParams, postOrder } from '../modules/order.ts'

export default connect(
  (state) => ({
    cart: state.cart.rows,
    requestParams: state.order.requestParams,
    postResultObj: state.order.postResultObj
  }),
  (dispatch) => ({
    setRequestParams: (addressForm) =>  dispatch(setRequestParams(addressForm)),
    postOrder: () =>  dispatch(postOrder())
  })
)(ConfirmOrder)