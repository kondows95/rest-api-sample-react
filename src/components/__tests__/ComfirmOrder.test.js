import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ConfirmOrder from '../ConfirmOrder';
import { Provider } from 'react-redux'
import store from '../../store';
import { createBrowserHistory } from 'history';
import { jsxEmptyExpression } from '@babel/types';
let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.outerHTML = '<body></body>'
  container = null
})

const valueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "value").set;
const requestParams = {
  first_name: "wint",
  last_name: "shwe yee",
  address1: "ye mon taung",
  address2: "mandalay",
  country: "myanmar",
  state: "mandalay",
  city: "mandalay"
}
const postResultObj = {
  id: 5,
  total_price: 50000
}
const history = { push: jest.fn() };
const postOrder =jest.fn(() => {
});
describe("Confirm Order component", () => {
  it('matches the snapshot', () => {
    const ComfirmOrderSnapshotWithPostObj = renderer.create(
      <Provider store={store}>
        <Router>
          <ConfirmOrder
            requestParams={requestParams}
            postResultObj={postResultObj}
          />
        </Router>
      </Provider>
    ).toJSON();
    expect(ComfirmOrderSnapshotWithPostObj).toMatchSnapshot();
  });
  it('matches the snapshot', () => {
    const ComfirmOrderSnapshot = renderer.create(
      <Provider store={store}>
        <Router>
          <ConfirmOrder
            requestParams={requestParams}
            postResultObj={null}
          />
        </Router>
      </Provider>
    ).toJSON();
    expect(ComfirmOrderSnapshot).toMatchSnapshot();
  });
});

describe("testing Checkout ", () => {
  it('testing checkout order confirm page', () => {
    act(() => {
      ReactDOM.render((
        <Provider store={store}>
          <Router>
            <ConfirmOrder
              requestParams={requestParams}
              postResultObj={null}
              history ={history}
              postOrder = {postOrder}
            />
          </Router>
        </Provider>
      ), container);
    });
    const spanArr = document.querySelectorAll('span');
    expect(spanArr[0].textContent.toUpperCase()).toBe('FIRST NAME');
    expect(spanArr[2].textContent.toUpperCase()).toBe('LAST NAME');
    expect(spanArr[4].textContent.toUpperCase()).toBe('ADDRESS1');
    expect(spanArr[6].textContent.toUpperCase()).toBe('ADDRESS2');
    expect(spanArr[8].textContent.toUpperCase()).toBe('COUNTRY');
    expect(spanArr[10].textContent.toUpperCase()).toBe('STATE');
    expect(spanArr[12].textContent.toUpperCase()).toBe('CITY');
    expect(spanArr[1].textContent).toBe('wint');
    expect(spanArr[3].textContent).toBe("shwe yee");
    expect(spanArr[5].textContent).toBe("ye mon taung");
    expect(spanArr[7].textContent).toBe("mandalay");
    expect(spanArr[9].textContent).toBe("myanmar");
    expect(spanArr[11].textContent).toBe("mandalay");
    expect(spanArr[13].textContent).toBe("mandalay");
    const buttonArr = document.querySelectorAll('Button');
    expect(buttonArr[0].textContent.toUpperCase()).toBe("GO PREVIOUS")
    expect(buttonArr[1].textContent.toUpperCase()).toBe("MAKE AN ORDER")
    act(() => {
      buttonArr[1].dispatchEvent(new Event('click', { bubbles: true }))
    });
    expect(postOrder).toHaveBeenCalled();
  })
  it('testing order success page', () => {
    act(() => {
      ReactDOM.render((
        <Provider store={store}>
          <Router>
            <ConfirmOrder
              requestParams={requestParams}
              postResultObj={postResultObj}
              history ={history}
              postOrder = {postOrder}
            />
          </Router>
        </Provider>
      ), container);
    });
    const spanArr = document.querySelectorAll('span');
    expect(spanArr[0].textContent).toBe("5");
    expect(spanArr[1].textContent.toUpperCase()).toBe("50000");
    
  });
})


