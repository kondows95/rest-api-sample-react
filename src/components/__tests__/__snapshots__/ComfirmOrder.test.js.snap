// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Order component matches the snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-7"
>
  <div
    className="MuiGrid-root"
  >
    <div
      className="MuiBox-root MuiBox-root-111"
    >
      Thank you for your order.
    </div>
    <div
      className="MuiBox-root MuiBox-root-112"
    >
      Thank you very much for your order.
    </div>
    <div
      className="MuiBox-root MuiBox-root-113"
    >
      Order contents have been sent through [order confirmation mail].
    </div>
    <div
      className="MuiBox-root MuiBox-root-114"
    >
      Please confirm the order contents, if there are any mismatches or unclear points, please contact us.
    </div>
    <div
      className="MuiBox-root MuiBox-root-115"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="MuiBox-root MuiBox-root-143"
        >
          Order Number
        </div>
        <div
          className="MuiBox-root MuiBox-root-144"
        >
          <span>
            5
          </span>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-145"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="MuiBox-root MuiBox-root-146"
        >
          Total Price
        </div>
        <div
          className="MuiBox-root MuiBox-root-147"
        >
          <span>
            50000
          </span>
           MMK
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-148"
    >
      <button
        className="MuiButtonBase-root MuiButton-root makeStyles-button-3 MuiButton-contained MuiButton-containedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-leftIcon-4"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
          Go To HOME
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Confirm Order component matches the snapshot 2`] = `
<div
  className="MuiBox-root MuiBox-root-188"
>
  <form
    className="makeStyles-form-1"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="MuiBox-root MuiBox-root-189"
    >
      1.Order contents
    </div>
    <div
      className="MuiBox-root MuiBox-root-190"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="MuiBox-root MuiBox-root-195"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4"
          >
            <div
              className="MuiBox-root MuiBox-root-196"
            >
               
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4"
          >
            <div
              className="MuiBox-root MuiBox-root-197"
            >
               SUBTOTAL 
            </div>
            <div
              className="MuiBox-root MuiBox-root-198"
            >
               SHIPPING COSTS 
            </div>
            <div
              className="MuiBox-root MuiBox-root-199"
            >
               ADDITIONAL SHIPPING COSTS
            </div>
            <div
              className="MuiBox-root MuiBox-root-200"
            >
              SALES TAX
            </div>
            <div
              className="MuiBox-root MuiBox-root-201"
            >
              TOTAL COSTS(4 ARTICLSE)
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4"
          >
            <div
              className="MuiBox-root MuiBox-root-202"
            >
               
              0
               MMK 
            </div>
            <div
              className="MuiBox-root MuiBox-root-203"
            >
               0.0 MMK 
            </div>
            <div
              className="MuiBox-root MuiBox-root-204"
            >
               0.0 MMK 
            </div>
            <div
              className="MuiBox-root MuiBox-root-205"
            >
               0.0 MMK 
            </div>
            <div
              className="MuiBox-root MuiBox-root-206"
            >
               
              0
               MMK
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-207"
    >
      2.Shipping Address
    </div>
    <div
      className="MuiBox-root MuiBox-root-208"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiBox-root MuiBox-root-209"
          >
            <div
              className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                className="MuiBox-root MuiBox-root-210"
              >
                <span>
                  First Name
                </span>
              </div>
              <div
                className="MuiBox-root MuiBox-root-211"
              >
                <span>
                  wint
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiBox-root MuiBox-root-212"
          >
            <div
              className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                className="MuiBox-root MuiBox-root-213"
              >
                <span>
                  Last Name
                </span>
              </div>
              <div
                className="MuiBox-root MuiBox-root-214"
              >
                <span>
                  shwe yee
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiBox-root MuiBox-root-215"
          >
            <div
              className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                className="MuiBox-root MuiBox-root-216"
              >
                <span>
                  Address1
                </span>
              </div>
              <div
                className="MuiBox-root MuiBox-root-217"
              >
                <span>
                  ye mon taung
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiBox-root MuiBox-root-218"
          >
            <div
              className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                className="MuiBox-root MuiBox-root-219"
              >
                <span>
                  Address2
                </span>
              </div>
              <div
                className="MuiBox-root MuiBox-root-220"
              >
                <span>
                  mandalay
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiBox-root MuiBox-root-221"
          >
            <div
              className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                className="MuiBox-root MuiBox-root-222"
              >
                <span>
                  Country
                </span>
              </div>
              <div
                className="MuiBox-root MuiBox-root-223"
              >
                <span>
                  myanmar
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiBox-root MuiBox-root-224"
          >
            <div
              className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                className="MuiBox-root MuiBox-root-225"
              >
                <span>
                  State
                </span>
              </div>
              <div
                className="MuiBox-root MuiBox-root-226"
              >
                <span>
                  mandalay
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiBox-root MuiBox-root-227"
          >
            <div
              className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                className="MuiBox-root MuiBox-root-228"
              >
                <span>
                  City
                </span>
              </div>
              <div
                className="MuiBox-root MuiBox-root-229"
              >
                <span>
                  mandalay
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-230"
    >
      3.Payment Method
    </div>
    <div
      className="MuiBox-root MuiBox-root-231"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="MuiBox-root MuiBox-root-232"
        >
          Credit Card Pay
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-233"
    >
      <div
        className="MuiBox-root MuiBox-root-234"
      >
        <button
          className="MuiButtonBase-root MuiButton-root makeStyles-button-3 MuiButton-contained MuiButton-containedPrimary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root makeStyles-leftIcon-4"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M14 7l-5 5 5 5V7z"
              />
              <path
                d="M24 0v24H0V0h24z"
                fill="none"
              />
            </svg>
            Go Previous
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root makeStyles-button-3 MuiButton-contained MuiButton-containedSecondary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Make an order
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root makeStyles-rightIcon-5"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 17l5-5-5-5v10z"
              />
              <path
                d="M0 24V0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;
